online-shop/src/app/page-product-details/confirmation-modal/confirmation-modal.component.ts

```diff
@@ -0,0 +1,27 @@
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+
+@Component({
+  selector: 'app-confirmation-modal',
+  templateUrl: './confirmation-modal.component.html',
+  styleUrls: ['./confirmation-modal.component.css']
+})
+export class ConfirmationModalComponent implements OnInit {
+  @Output() public confirm: EventEmitter<any> = new EventEmitter<any>();
+  modalTextValue: string;
+  @Input() modalId: string;
+
+  @Input()
+  set modalText(value) {
```

Comment:
Why do you need a setter in addition to the modalTextValue property? Why not directly use the property as an input?